SonarQube.Client/Api/V9_4/GetSonarLintEventStream.cs

```diff
@@ -42,6 +46,42 @@ internal class GetSonarLintEventStream : RequestBase<Stream>, IGetSonarLintEvent
                 MediaTypeWithQualityHeaderValue.Parse("text/event-stream")
             };
 
+        public override async Task<Stream> InvokeAsync(HttpClient httpClient, CancellationToken token)
+        {
+            var requestAttempts = 0;
+
+            return await InvokeWithRetriesAsync();
+
+            // Code on the java side: https://github.com/SonarSource/sonarlint-core/blob/4f34c7c844b12e331a61c63ad7105acac41d2efd/server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/stream/EventStream.java#L101
+            async Task<Stream> InvokeWithRetriesAsync()
+            {
+                do
+                {
+                    requestAttempts++;
+
+                    var result = await InvokeUncheckedAsync(httpClient, token);
+
+                    if (ShouldRetry(result) && requestAttempts < MaxNumberOfRequestAttempts)
+                    {
+                        continue;
+                    }
+
+                    result.EnsureSuccess();
+
+                    return result.Value;
+                } while (true);
+            }
+        }
+
+        private static bool ShouldRetry(Result<Stream> result)
+        {
+            var isRecoverableStatus = result.StatusCode != HttpStatusCode.Forbidden &&
```

Comment:
Make this an extension method maybe? It would not add much value in terms of reusing the code, but it would look nicer

Reply:
Hmm, it will need to be private since it's only unique to this API, I don't see us reusing it anywhere.

Reply:
Extension methods can't be private or in nested classes, so we couldn't really hide it properly. I don't think it would be an improvement overall.